main.js

```diff
@@ -23,41 +20,58 @@ hlDefault.use(new defaultCodecs.MultihashSha2256());
 hlDefault.use(new defaultCodecs.MultihashBlake2b64());
 hlDefault.use(new defaultCodecs.MultibaseBase58btc());
 
+const DEFAULT_CODEC_IDS = ['mh-sha2-256', 'mb-base58-btc'];
+
 /**
  * Encodes a hashlink. If only a `url` parameter is provided, the URL is
  * fetched, transformed, and encoded into a hashlink. If a data parameter
  * is provided, the hashlink is encoded from the data.
  *
- * @param {Object} options - The options for the encode operation.
+ * @param {object} options
  * @param {Uint8Array} options.data - The data associated with the given URL. If
  *   provided, this data is used to encode the cryptographic hash.
- * @param {Array} options.urls - One or more URLs that contain the data
- *   referred to by the hashlink.
- * @param {Array} options.codecs - One or more URLs that contain the data
- *   referred to by the hashlink.
- * @param {Object} options.meta - A set of key-value metadata that will be
+ * @param {string|Array<string>} options.url - One or more URLs that resolve to
+ *   the data referred to by the hashlink.
+ * @param {Array} [options.codecs] - One or more codecs that should be used
+ *   to encode the data.
+ * @param {object} [options.meta={}] - A set of key-value metadata that will be
  *   encoded into the hashlink.
  *
  * @returns {Promise<string>} Resolves to a string that is a hashlink.
  */
-async function encode({data, urls, url,
-  codecs = ['mh-sha2-256', 'mb-base58-btc'], meta = {}}) {
+async function encode(options) {
+  let data, url, codecs, meta;
+
+  if(!options) {
+    throw new Error('Urls and/or data params are required.');
+  }
 
-  if(url && !urls) {
-    urls = [url];
+  if(typeof options === 'string') { // Convenience usage, `create(url)`
+    url = options;
+    codecs = DEFAULT_CODEC_IDS;
+    meta = {};
+  } else {
+    data = options.data;
+    url = options.url;
+    codecs = options.codecs || DEFAULT_CODEC_IDS;
+    meta = options.meta || {};
   }
 
-  return await hlDefault.encode({data, urls, codecs, meta});
+  if(!data) {
+    // TODO: Add fetching of data from url
+    throw new Error('Fetching of data from urls coming soon.');
```

Comment:
I think "coming soon" should be left out.  Better to just say what the problem is:
```suggestion
    throw new Error('Fetching of data from urls not implemented.');
```